client/src/Views/Home/Prospects/Prospects.js

```diff
@@ -11,151 +18,305 @@ import ProspectTableCheckbox from "Components/Checkbox/ProspectTableCheckbox";
 
 import ProspectSidebar from "./ProspectSidebar";
 import ProspectDashboardHeader from "./ProspectDashboardHeader";
+import CampaignSelectDialog from "Views/Home/Prospects/CampaignSelect";
+import { addProspectsToCampaign } from "Utils/api";
 
 class Prospects extends Component {
-  
   constructor(props) {
     super(props);
     this.state = {
-      prospects : [],
+      prospects: [],
       filteredProspects: [],
-      user : {},
-      checked : false,
+      prospectChecked: {}, // tracks which prospectIds have checkboxes checked in table
+      user: {},
+      allChecked: false,
       errors: {},
-    }
-  }
-  
-
-  handleClick = (event, id) => {
-    console.log(event);
+      addToCampaignBtnVisible: false,
+      showCampaignDialog: false
+    };
   }
-  
 
   componentDidMount = async () => {
     try {
       this.state.user = this.context.user;
       const res = await getProspectData(this.state.user.id);
+      const prospectChecked = {};
+
+      res.data.forEach(prospect => {
+        prospectChecked[prospect._id] = false;
+      });
+
       this.setState({
         prospects: res.data,
-        filteredProspects : res.data
+        filteredProspects: res.data,
+        prospectChecked: { ...prospectChecked }
       });
-    } catch(error) {
+    } catch (error) {
       this.setState({
         errors: { error }
       });
     }
-  }
+  };
 
-  handleChange = (event) => {
-    this.setState({ ...this.state, [event.target.name]: event.target.checked });
+  handleChange = (event, prospectId) => {
+    // if 'check all' checkbox was selected ...
+    if (prospectId === "all") {
```

Comment:
I don't mind setting this variable to all if the header is selected, but it does make it harder to read - I would have favoured creating a separate function for that job